### PR TITLE
[rfw] Material slider widget

### PR DESCRIPTION
**Pull request by uberchilly**
### Description
This adds Slider widget to material widgets.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/rfw/lib/src/flutter/material_widgets.dart
Comment (Hixie) [inline packages/rfw/lib/src/flutter/material_widgets.dart:515]: Per the [style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo), indentation should never have a line that is less indented than a line that is of higher lexical scope (well the style guide doesn't say it quite that way but that's the intent).
```suggestion
      onChanged: source.handler(['onChanged'],
        (HandlerTrigger trigger) => (double value) {
          trigger({'value': value});
        },
      ),
```

Comment (Hixie) [reply packages/rfw/lib/src/flutter/material_widgets.dart:515]: (same below)

File: packages/rfw/test/utils.dart
Comment (Hixie) [inline packages/rfw/test/utils.dart:27]: can you make this a function instead of an extension? that would make it easier to find when reading the code (otherwise it looks like you have to look on WidgetTester).

See also https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-extension

**Comment by Hixie** [comment]
This looks great! Sorry it took me so long to review! Just some minor nits.

**Comment by uberchilly** [comment]
Please let me know if there is something else I can fix after latest changes

**Review by Hixie — APPROVED** [review]
looks great, thanks!
LGTM

**Comment by uberchilly** [comment]
Thank you for the approval. Do we need to wait for another approval before this ends up in version 1.0.27 or how does it work from here?

**Comment by Hixie** [comment]
Oh I forgot it needed a second review. Let me post about it in the Discord. Sorry about that. Yay for the bots!

File: packages/rfw/lib/src/flutter/material_widgets.dart
Comment (ditman) [inline packages/rfw/lib/src/flutter/material_widgets.dart:508]: Should the number of decimals be configurable from the definition as well? I think having a "0 decimals" which calls `round` might be a very common use case.

**Review by ditman — APPROVED** [review]
LGTM, small question about the configurability of the "label" /cc @Hixie

File: packages/rfw/lib/src/flutter/material_widgets.dart
Comment (uberchilly) [reply packages/rfw/lib/src/flutter/material_widgets.dart:508]: Btw technically speaking, since we cannot really "script" from outside, easily having label as string param is not useful since one can only provide hardcoded text that is why I've also added common use case of showing current places with 2 decimal places but I can also add number of decimal places as a param.

Comment (ditman) [reply packages/rfw/lib/src/flutter/material_widgets.dart:508]: > having label as string param is not useful since one can only provide hardcoded text

@uberchilly I agree. I also think the current solution is not great for i18n. I'd suggest you did this with something similar to `sprintf` or a way for users to specify where in the string they want the number (to enable them to pass `'## Birds'` so the label is "18 Birds" rather than "Birds: 18").

> I've also added common use case of showing current value with 2 decimal places but I can also add number of decimal places as a param.

I don't think 2 decimal places is more common than 1 decimal place or 0 (or 5), it all depends on the number of "divisions" or the size of the increment. For example, if I were to use this widget with integer values (between 0 and 100 with a step of 10), I think I wouldn't want to see *any* decimal positions anywhere.

If I was doing between 0 to 1 with 1000 divisions, I'd certainly would want to see increments in the 3rd decimal.

Instead of attempting to figure out what's the ideal number of decimals, I'd just add a value so users can configure it (and that's why I suggested it).

Comment (uberchilly) [reply packages/rfw/lib/src/flutter/material_widgets.dart:508]: I agree in general, but on the other hand other already supported widgets have bigger limitations imo. In my personal project because of these limitations I had to also add DoubleText and IntText as local widgets because I couldn't do what was suggested here somewhere and that was to add string variant of value that I need to show in Text in data beside regular double for example, and showing number value in Text widget is much more common usecase than using label in slider. So, I am not sure how deep should each widget go in terms of supporting things vs an effort to add scripting easily to prepare values from outside

**Comment by ditman** [comment]
@stuartmorgan I'm not opposed to this landing!

**Pull request merged**
